tests/src/unit-bjdata.cpp

```diff
@@ -2094,6 +2094,20 @@ TEST_CASE("BJData")
             CHECK(!json::sax_parse(v, &scp, json::input_format_t::bjdata));
         }
 
+        SECTION("key() in ndarray _ArrayType_")
+        {
+            std::vector<uint8_t> v = {'[', '$', 'U', '#', '[', '$', 'U', '#', 'i', 2, 2, 2, 1, 2, 3, 4};
+            SaxCountdown scp(9);
+            CHECK(!json::sax_parse(v, &scp, json::input_format_t::bjdata));
+        }
+
+        SECTION("key() in ndarray _ArrayType_")
+        {
+            std::vector<uint8_t> v = {'[', '$', 'U', '#', '[', '$', 'U', '#', 'i', 2, 2, 2, 1, 2, 3, 4};
+            SaxCountdown scp(10);
+            CHECK(!json::sax_parse(v, &scp, json::input_format_t::bjdata));
+        }
+
```

Comment:
From my testing, the correct counts seem to be 6 and 7.

Please also replace all occurrences of `CHECK(!` with `CHECK_FALSE(`.

Reply:
@falbrechtskirchinger, thanks. is there a way to print the event count? I previously try to many count but sometime my count does not match the expected.

Reply:
Not that I know off, I just step through it with the debugger.

Reply:
Did I give you the wrong line, or what's going on here?

https://coveralls.io/builds/48991323/source?filename=include%2Fnlohmann%2Fdetail%2Finput%2Fbinary_reader.hpp#L2456

Line 2456 is being missed now. I'm going to compare it against the previous coverage run.

Reply:
No, it was definitely line 2445 before.

https://coveralls.io/builds/48971838/source?filename=include%2Fnlohmann%2Fdetail%2Finput%2Fbinary_reader.hpp#L2445

:man_shrugging:

Reply:
> @falbrechtskirchinger, thanks. is there a way to print the event count? I previously try to manually count but sometime my count does not match the expected.

Do you mean something like https://github.com/doctest/doctest/blob/master/doc/markdown/logging.md#info?

Reply:
I guess we could add `CHECK(m_event_count == -1)` to `~SaxEventCountdown()`. But that wouldn't solve things completely. A more advanced solution could add an enum to indicate in which event we expect the counter to reach `-1` and check if that happened in the destructor. I'll add that to my to-do list. There was another PR that moved `SaxEventCountdown` into a separate header. Probably a good idea as well.